wgsl/index.bs

```diff
@@ -683,16 +684,80 @@ module header and each usage of a GLSL method will add the appropriate
   </xmp>
 </div>
 
+## Module Constants ## {#module-constants}
+
+A *module constant* declares a name for a value, outside of all function declarations.
+The name is available for use after the end of the declaration,
+until the end of the [SHORTNAME] program.
+
+When the declaration has no attributes, an initializer expression must be present,
+and the name denotes the value of that expression.
+
+<div class='example' heading='Module constants'>
+  <xmp>
+    const golden : f32 = 1.61803398875;       # The golden ratio
+    const e2 : vec3<i32> = vec3<i32>(0,1,0);  # The second unit vector for three dimensions.
+  </xmp>
+</div>
+
+When the declaration uses the `constant_id` attribute,
+the constant is *pipeline-overridable*. In this case:
+
+  * The type must one of the [[#scalar-types]].
+  * The initializer expression is optional.
+  * The attribute's literal operand is known as the *pipeline constant ID*,
+    and must be a non-negative integer value representable in 32 bits.
+  * Pipeline constant IDs must be unique within the [SHORTNAME] program: Two module constants
+    must not use the same pipeline constant ID.
+  * The application can specify its own value for the name at pipeline-creation time.
+    The pipeline creation API accepts a mapping from the pipeline constant ID
+    to a value of the constant's type.
+    If the mapping has an entry for the ID, the value in the mapping is used.
+    Otherwise, the initializer expression must be present, and its value is used.
```

Comment:
This was the simplest way to say it. 
But the implementation detects the error at pipeline creation time, not at shader module creation time.   It's pipeline creation that fails, no earlier.